detectron2/layers/csrc/cuda_version.cu

```diff
@@ -4,6 +4,10 @@
 
 namespace detectron2 {
 int get_cudart_version() {
+#if __HIP_PLATFORM_NVCC__ 
```

Comment:
When compiled with regular NVCC, does this macro exist?

Reply:
Done. Also modified collect_env.py to print if the GPU platform is CUDA or HIP, and what version etc.